src/Propel/Generator/Reverse/PgsqlSchemaParser.php

```diff
@@ -417,7 +417,10 @@ protected function addForeignKeys(Table $table, $oid)
                 $fk = new ForeignKey($name);
                 $fk->setForeignTableCommonName($foreignTable->getCommonName());
                 if ($table->guessSchemaName() != $foreignTable->guessSchemaName()) {
-                    $fk->setForeignSchemaName($foreignTable->getSchema());
+                     if($foreignTable->getSchema())
+                            $fk->setForeignSchemaName($foreignTable->getSchema());
+                     elseif($foreignTable->getSchema() == 'public')
```

Comment:
As stated by the Postgres documentation:

_tables created without specifying any schema names, by default such tables (and other objects) are automatically put into a schema named "public". Every new database contains such a schema_

Here is the link: https://www.postgresql.org/docs/9.1/static/ddl-schemas.html Section 5.7.2